force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +125,21 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        Integer results = 0; 
+
+        //  get individual characters from the string - hint use substring or split
+        // check if the character is a vowel
+        // if it is a vowel, increment the result Integer
+
+        for (Integer i = 0; i < str.length(); i++) {
+            String individualCharacter = str.substring(i, i + 1);
+            if (individualCharacter == 'a' || individualCharacter == 'e' || individualCharacter == 'i' || individualCharacter =='o' || individualCharacter =='u'){
+                results = results + 1;
+               
```

Comment:
This works. But lots of conditional logic is hard to follow and hard to test 5 OR conditions  could be replaced with one call to a Set using the contains method. 

```
 public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```